docs/v1-proposed-api.md

```diff
@@ -0,0 +1,254 @@
+   * BaseDomain: Used to drive the automatic creation of the DNSZone if manageDNS is true. May be removed in V2.
+ 1. Introduce Spec.ClusterMetadata struct with fields for InfraID, ClusterUUID, and admin password/kubeconfig secret references.
+   * All of these fields will be removed from Status where they cannot be restored.
+   * InfraID and ClusterUUID will be programatically mutable, but once set and the cluster is marked installed, validation will make them immutable.
+   * ClusterMetadata may be renamed if a better idea comes up.
+ 1. Rename ClusterDeployment.Spec.SSHKey to ClusterDeployment.Spec.SSHKeySecret to better match all other secret references.
+   * Should not be removed as private keys are not contained in InstallConfig and are used by Hive.
+   * Rename to SSHPrivateKeySecret and should only contain private key going forward.
+ 1. Rename CertificateBundle.SecretRef to CertificateSecret.
+ 1. Move PlatformSecrets into Platform, rename Credentials to CredentialsSecret.
+ 1. Add support for bare metal provisioning.
+   * At this point, we do not see any major implications for BareMetal. This is modelled as a separate platform in install config (see [this example](https://github.com/openshift/installer/blob/master/docs/user/metal/install_ipi.md#install-config))
+   * This would be specified in the InstallConfig.
+   * Hive would likely want a platform indicator explicitly for bare metal if we need a special invocation of the installer.
+   * However for the most part, we do not see any major complications adding bare metal support to the Hive API.
+ 1. Add support for hosted control plane provisioning. (control plane running in Hive management cluster, in the same namespace as the ClusterDeployment, but cluster workers in a remote cloud provider account)
+   * A lot needs to be settled here before we can know exactly how this would look. However looking over the [current inputs for the POC code](https://github.com/sjenning/hosted-cluster-poc/blob/master/config.sh.aws_example), we will likely combine a Spec.Platform as we have today, with another Spec.HostedControlPlane section containing a variety of settings.
+   * May deploy control plane to the same cluster as Hive, or give Hive a kubeconfig for a remote cluster where the control plane should run.
+   * Safer to assume we are not the cluster where control plane will run.
+   * Spec.Platform.AWS specified, as well as Spec.HostedControlPlane.Kubeconfig etc.
+
+
+### Proposed V1 ClusterDeployment
+
+```yaml
+apiVersion: hive.openshift.io/v1alpha1
+kind: ClusterDeployment
+metadata:
+  name: ci-cluster-v4-1
+  namespace: uhc-staging-18urm2sbab85d9crlk27is39n706ph3g
+spec:
+  baseDomain: r4f1.s1.devshift.org
+  certificateBundles:
+  - generate: true
+    name: primary-cert-bundle
+    certificateSecret:
+      name: primary-cert-bundle-secret
+  compute: # controlPlane is gone, install config only, but may reappear here if these fall under machine set management in future.
```

Comment:
If masters become machinesets would it make more sense to name this `machineSets` instead of `compute`?